datadog/fwprovider/resource_datadog_integration_aws_event_bridge.go

```diff
@@ -0,0 +1,249 @@
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	resp, httpResp, err := r.Api.ListAWSEventBridgeSources(r.Auth)
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "Error listing AWS Event Bridge Sources"))
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		response.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+
+	id := state.ID.ValueString()
+
+	found := false
+	matchedEventHub := integrationAwsEventBridgeModel{}
+
+	if accounts, ok := resp.GetAccountsOk(); ok && len(*accounts) > 0 {
+		for _, account := range *accounts {
+			if found {
+				break
+			}
+			if eventhubs, ok := account.GetEventHubsOk(); ok && len(*eventhubs) > 0 {
+				for _, eventhub := range *eventhubs {
+					if *eventhub.Name == id {
+						matchedEventHub.ID = types.StringValue(*eventhub.Name)
+						matchedEventHub.AccountId = types.StringValue(account.GetAccountId())
+						matchedEventHub.Region = types.StringValue(eventhub.GetRegion())
+						matchedEventHub.EventGeneratorName = types.StringValue(eventhub.GetName())
+						found = true
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if !found {
+		response.State.RemoveResource(ctx)
+		response.Diagnostics.AddError("Import error", "No matching Event Source found")
+		return
+	}
+
+	r.updateStateAfterRead(ctx, &state, &matchedEventHub)
+	// Save data into Terraform state
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *integrationAwsEventBridgeResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var state integrationAwsEventBridgeModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	integrationAWSEventBridgeMutex.Lock()
+	defer integrationAWSEventBridgeMutex.Unlock()
+
+	body, diags := r.buildIntegrationAwsEventBridgeRequestBody(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resp, _, err := r.Api.CreateAWSEventBridgeSource(r.Auth, *body)
+	if err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "Error creating AWS Event Bridge Source"))
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		response.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+	r.updateStateAfterWrite(ctx, &state, &resp)
+
+	// Save data into Terraform state
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *integrationAwsEventBridgeResource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	response.Diagnostics.AddError("Update not supported", "AWS Event Bridge Sources cannot be updated")
```

Comment:
@skarimo Is this ok to do? There is no way to update this resource via the API and we don't intend to support updates. Let me know if there's another way this should be handled

Reply:
Yes, this is good! Im going to push up a small change which should ensure we never hit this codepath. And if we were to hit it somehow, erroring out is the right choice